spec/konfa_spec.rb

```diff
@@ -159,6 +159,17 @@ def allowed_variables
     it 'should return self' do
       expect(MyTestKonfa.init_with(:something)).to eq(MyTestKonfa)
     end
+
+    it 'sets initialized to false' do
+      expect(MyTestKonfa).to receive(:reinit)
+      MyTestKonfa.init_with(:something)
+    end
+
+    it 'subsequent calls to init_with do not require reinit' do
+      MyTestKonfa.init_with(:yaml, good_file).init
+      expect(MyTestKonfa).to receive(:after_initialize)
```

Comment:
This is a bit of a subtle way to test it but since after_initialize returns `nil` I'd have to subclass and override it to be able to use `expect(MyTestKonfa.init_with(:yaml, good_file).init).not_to be_nil` instead. What do you think @ghansson?